### ISSUE TITLE
[Exchange] Cancel order option in BUY/SELL ORDERS

### ISSUE THREAD

**Issue by froooze** [bitshares/bitshares-ui#2563]
**Is your feature request related to a problem? Please describe.**
Scrolling through order books and cancel a order needs a switch to MY OPEN ORDERS.

**Describe the solution you'd like**
Add the cancel order icon also to the BUY/SELL ORDERS afterwards.

**Describe alternatives you've considered**
Only make a right-click context menu to cancel the order in BUY/SELL ORDERS.

**Comment by startailcoon** [comment]
> Add the cancel order icon also to the BUY/SELL ORDERS afterwards.

Does this imply that you want a single button to cancel all your open orders on said market pair?

**Comment by froooze** [comment]
It was not intended, but we can include check boxes with one master box on top for check and uncheck all orders of one asset.

#2317

**Comment by startailcoon** [comment]
Is issue #2317 what your looking for? Then we can close this issue.

**Comment by froooze** [comment]
When #2317 gets integrated in the order book you can close this.

**Comment by startailcoon** [comment]
You mean you wish to have a cancel option directly in the orderbook where you have orders placed?

**Comment by froooze** [comment]
Yes, how about a right click context menu to cancel the order, which doesn't take extra space?
Scrolling through order books, need a simple and direct way to cancel a specific order, without removing the focus. 

To cancel multiple orders is more important for MY OPEN ORDERS, than in the BUY/SELL ORDERS.

**Comment by startailcoon** [comment]
I'm not sure what your request is when you state that it's **more** important in `My Open Orders` than in the `Buy/Sell Orderbook`. This was resolved by #2317. 

Can we consider this closed?

**Comment by froooze** [comment]
No, because this ticket is about a right click option for BUY/SELL ORDERS to cancel one order!

**Comment by startailcoon** [comment]
Sorry for the misunderstanding, now I understand your requirements :) 

When a user has a position in the order book, enable right click to cancel all positions on that price that belongs to the user.
- [ ] Add tooltip to inform the user when hovering on a position in the order book that can be canceled.
- [ ] Add "right click" action to fire a cancel action. (Left click is used to send price to the form)

**Comment by OpenLedgerApp** [comment]
@startailcoon, I can take this one.

**Comment by froooze** [comment]
@startailcoon: We should not overdo it with tooltips, I found them recently annoying, by blocking important UI elements.

**Comment by abitmore** [comment]
How to do "right click" on a phone? Long press? How to do "hover"? Tap it? When designing please always remember that bitshares-ui is not for PC only. Thanks.

**Comment by startailcoon** [comment]
How I see it, we could do this in two different ways

1. A icon, as we have on open orders. This is not very practical as not all orders can be canceled by the user. It would require some space, that we may not have.
2. A right click or long press to open a popup menu to enable cancel.

Some questions that arise are
- Can we detect a long press for mobile?
- How does one know there is a right click or long press action?

**Comment by froooze** [comment]
The only right click function we have now is copy? 
The right click menu keeps the UI clean and provides the tools for power user. 

We could mention it in the start documentation?

**Comment by startailcoon** [comment]
@OpenLedgerApp , any ideas or inputs as you're interested in working on this issue? Perhaps #2700 can give some ideas? The issue we have is how to handle mobile devices as they don't do hover.

**Comment by froooze** [comment]
@startailcoon: Power user will find out fast if the right click menu has some useful tools. Everything we add to the UI makes it not better.

**Comment by sschiessl-bcp** [comment]
> @OpenLedgerApp , any ideas or inputs as you're interested in working on this issue? Perhaps #2700 can give some ideas? The issue we have is how to handle mobile devices as they don't do hover.

The mobile view is a mess and sluggish .. I would rather have two components (that reuse the functional components) to be honest instead of this one monster.

What I mean, don't be influenced too much by the need of responsive design.

**Comment by startailcoon** [comment]
> What I mean, don't be influenced too much by the need of responsive design. Typically, swiping left or right would unlock deleting on phones.

Good point. Still, how can we solve this on desktop if we set mobile functionality aside for the moment?
